src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/config/InterceptorConfig.java

```diff
@@ -24,5 +31,9 @@ public InterceptorConfig(LoggingInterceptor loggingInterceptor) {
     @Override
     public void addInterceptors(@NonNull InterceptorRegistry registry) {
         registry.addInterceptor(loggingInterceptor);
+        registry.addInterceptor(new TokenPermissionsInterceptor());
+        registry.addInterceptor(new UserAuthenticationInterceptor(Collections.EMPTY_LIST,
```

Comment:
Rather than calling the token permission interceptor, which just adds tokens to a request, it looks like we require the CRUD interceptor, which seems to unpack and check permissions at the other end when a request is received. Here we can pass in COMPANY_INCORPORATION. If we add this together with the auth interceptor
then wee seem to do the three things that the roe auth interceptor does namely:

Check already logged in 
SecurityConstants.API_KEY_IDENTITY_TYPE.equals(AuthorisationUtil.getAuthorisedIdentityType(request))

Check token permissions
final var tokenPermissions = getTokenPermissions(request)
                .orElseThrow(() -> new IllegalStateException("UserAuthenticationInterceptor - TokenPermissions object not present in request"));

check company has permission
 boolean hasCompanyIncorporationCreatePermission = tokenPermissions.hasPermission(Key.COMPANY_INCORPORATION, Value.CREATE);

Reply:
https://github.com/companieshouse/overseas-entities-api/commit/1f3dc3f1ad87b23f27d7727370c4e50ddb3b7580#diff-bc2a5bfa0c1d43cb2e277e66d62a70754a6b5261872c009e9aa141bf92435fb3R38

Went from using CRUD to a custom interceptor presumably because we wanted log out the transaction ID, that's the only difference I see, should we then do the same here?

Reply:
Is was causing an error on accept and pay when the transaction was closed.